drivers/video/mt9m114.c

```diff
@@ -461,12 +503,35 @@ static int mt9m114_get_caps(const struct device *dev, enum video_endpoint_id ep,
 	return 0;
 }
 
+static int mt9m114_set_ctrl(const struct device *dev,
+				unsigned int cid, void *value)
+{
+	int ret = 0;
+
+	switch (cid) {
+	case VIDEO_CID_HFLIP:
+		ret |= mt9m114_set_horizontal_mirror(dev, (int)value);
+		break;
+	case VIDEO_CID_VFLIP:
+		ret |= mt9m114_set_vertical_flip(dev, (int)value);
+		break;
+	default:
+		return -ENOTSUP;
+	}
+
+	/* Apply Config */
+	ret |= mt9m114_set_state(dev, MT9M114_SYS_STATE_ENTER_CONFIG_CHANGE);
```

Comment:
It is true that the `ret |= ...` is everywhere in sensor drivers, but it might introduce small bugs in error handling. I.e. `-EIO | -EAGAIN == -EPERM` (`-5 | -11 == -1`).

It would be almost as concise to do it that way:

```suggestion
	switch (cid) {
	case VIDEO_CID_HFLIP:
		ret = mt9m114_set_horizontal_mirror(dev, (int)value);
		break;
	case VIDEO_CID_VFLIP:
		ret = mt9m114_set_vertical_flip(dev, (int)value);
		break;
	default:
		return -ENOTSUP;
	}
	if (ret < 0) {
		return ret;
	}

	/* Apply Config */
	return mt9m114_set_state(dev, MT9M114_SYS_STATE_ENTER_CONFIG_CHANGE);
```

Is it reasonable enough to switch?

An alternative is to convert all sensor drivers at once in a big refactoring when an opportunity shows-up (i.e. new APIs requiring modification anyway)...